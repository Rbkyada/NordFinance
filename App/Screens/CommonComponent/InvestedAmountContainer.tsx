import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';
import AppImages from '@Theme/AppImages';
import LightTheme from '@Theme/LightTheme';
import { CategoryTitle } from '@SubComponents/CategoryTitle';
import { BadgeBox } from '@SubComponents/BadgeBox';
import { en } from '@Localization/index';

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  boxContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  constInputStyle: {
    textAlign: 'center',
    paddingHorizontal: 10,
    paddingVertical: 2,
    marginRight: 10,
    minWidth: 50,
    borderRadius: 12,
    overflow: 'hidden',
    maxWidth: 80,
  },
});

interface InvestedAmountContainerProps {
  value: number;
  setValue: (e: number) => void;
}

const InvestedAmountContainer = (props: InvestedAmountContainerProps) => {
  const { value, setValue } = props;

  console.log('value', value);

  const { container, boxContainer, constInputStyle } = styles;

  return (
    <View style={container}>
      <CategoryTitle title={en.INVESTED_AMOUNT} />
      <View style={boxContainer}>
        <TextInput
          style={[constInputStyle, { backgroundColor: LightTheme.border }]}
          maxLength={7}
          value={(isNaN(value) && '') || value.toString()}
          onChangeText={e => setValue(+e)}
          keyboardType="numeric"
          returnKeyType="done"
        />
        <BadgeBox
          title={en.USD}
          isImg={AppImages.icArrowDown}
          textStyle={{ color: LightTheme.text }}
        />
      </View>
    </View>
  );
};

export { InvestedAmountContainer };
