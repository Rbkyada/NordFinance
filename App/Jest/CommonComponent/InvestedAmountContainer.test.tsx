import renderer from 'react-test-renderer';
import { TextInput } from 'react-native';
import { InvestedAmountContainer } from '@CommonComponent/InvestedAmountContainer';

jest.mock('react-native-config', () => () => ({ run: jest.fn() }));

describe('InvestedAmountContainer', () => {
  it('renders correctly', () => {
    const mockValue = 1000;
    const mockSetValue = jest.fn();

    const tree = renderer
      .create(
        <InvestedAmountContainer value={mockValue} setValue={mockSetValue} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls setValue with the parsed number when onChangeText is called', () => {
    const mockValue = 1000;
    const mockSetValue = jest.fn();
    const component = renderer.create(
      <InvestedAmountContainer value={mockValue} setValue={mockSetValue} />,
    );
    const textInput = component.root.findByType(TextInput);

    const newValue = '5000';
    textInput.props.onChangeText(newValue);

    expect(mockSetValue).toHaveBeenCalledWith(5000);
  });

  it('calls setValue with NaN when an invalid value is entered', () => {
    const mockValue = 1000;
    const mockSetValue = jest.fn();
    const component = renderer.create(
      <InvestedAmountContainer value={mockValue} setValue={mockSetValue} />,
    );
    const textInput = component.root.findByType(TextInput);

    const newValue = 'abc';
    textInput.props.onChangeText(newValue);

    expect(mockSetValue).toHaveBeenCalledWith(NaN);
  });
});
