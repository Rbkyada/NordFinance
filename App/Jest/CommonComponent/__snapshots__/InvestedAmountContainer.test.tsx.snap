// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestedAmountContainer renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 20,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      [
        {
          "fontSize": 18,
        },
        {
          "color": "#1c1c1e",
        },
        {
          "fontFamily": "DMSans-Regular",
        },
        [
          {
            "fontFamily": "DMSans-Bold",
          },
          {
            "color": "rgba(158, 172, 219, 1)",
          },
          undefined,
        ],
      ]
    }
    xlarge={true}
  >
    Invested Amount
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      maxLength={7}
      onChangeText={[Function]}
      returnKeyType="done"
      style={
        [
          {
            "borderRadius": 12,
            "marginRight": 10,
            "maxWidth": 80,
            "minWidth": 50,
            "overflow": "hidden",
            "paddingHorizontal": 16,
            "paddingVertical": 2,
            "textAlign": "center",
          },
          {
            "backgroundColor": "rgba(243, 243, 243, 1)",
          },
        ]
      }
      value="1000"
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "borderRadius": 12,
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "paddingVertical": 4,
          },
          {
            "backgroundColor": "rgba(243, 243, 243, 1)",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={0}
        small={true}
        style={
          [
            {
              "fontSize": 12,
            },
            {
              "color": "#1c1c1e",
            },
            {
              "fontFamily": "DMSans-Regular",
            },
            [
              {
                "color": "rgba(37, 42, 72, 0.2)",
              },
              [
                {
                  "paddingVertical": 4,
                },
                {
                  "color": "#1c1c1e",
                },
              ],
            ],
          ]
        }
      >
        USD
      </Text>
      <Image
        resizeMode="contain"
        source={
          {
            "uri": "arrow_down",
          }
        }
        style={
          {
            "height": 10,
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
